### PR TITLE
WIP: new data: shield, delivery console

### PR DESCRIPTION
**Pull request by gregglind**

### CODE REVIEW

**Comment by gregglind** [comment]
Goals:

- try to untangle shield, delivery console
- unify both under these concepts
  - new evidence
  - evidence-driven path to 100% of release for good ideas
- mitigating risk
- futher talk about deploying new probes in general.

Feedback wanted:
- do these framings make sense?
- am I boiling the ocean
- structure of document tree

Worries:
- this feels both closely and loosely aligned with the rest of the docs.

**Comment by gregglind** [comment]
I am making a lot of repetitive sections.  I am not sure how to approach various flowcharts and wizards, so submitting them all.  We can rip things down.

**Comment by gregglind** [comment]
@harterrt 

Proposed and questions:
1.  Proposed. top level '/experiments/' dir.
2. Observation. current state:  people confuse Shield as evidence gathering tool, and 'off-train deployment'.  Proposed: to  untangle those
3.  Proposed editorial. a 'user-problem' focused SUMMARY.md.  "I need help.", etc.  Claim: read taxonomy doesn't match our internal mapping.  Everything is an 'experiment'.  Further down in the summary is a 'reference' list.
4.  Question:  'introduction.md'  The rest of the book is about being an Analyst.  Most of the experimenting stuff is about Needing To Make a Decision / being a PM.
5. Question:  I have a lot of 'wizards' / and 'case blocks'.  Mermaid?  Seem heavy and hard and weird.  Examples:  "what do I need based on the evidence I want to collect / the decision I need to make".

**Comment by harterrt** [comment]
Thanks, Gregg. I'm really excited for this documentation! Some notes inline:
> 1.  Proposed. top level '/experiments/' dir.

There's two ways to view the documentation: as the source is stored and as it's rendered. The source is needs to make sense to folks *writing* documentation. I think it makes sense to have a separate `experiments/` top-level directory since these tools are maintained somewhat distinctly from the rest of the pipeline.

In fact, I think it makes sense for parts of the rendered documentation to be segregated as well (set off with a hbreak in SUMMARY.md). As a guiding principle, we want the main section of the documentation to be readable. As in, a new analyst should be able to start from the beginning and continue reading until they feel confident. Some of this new documentation is reference documentation (should be segregated from the main section) other articles are good introductions that every analyst coming to Mozilla should read (should be part of the main section). It looks like you already started making this distinction (e.g. "Experimentation using Shield" in the intro)

> 2. Observation. current state:  people confuse Shield as evidence gathering tool, and 'off-train deployment'.  Proposed: to  untangle those

Targeted lessons like these are good candidates for cookbooks. It may be work starting with a PR that just introduces this article to the cookbooks section. That way we can start referencing it immediately while you work on the remaining docs.

> 3.  Proposed editorial. a 'user-problem' focused SUMMARY.md.  "I need help.", etc.  Claim: read taxonomy doesn't match our internal mapping.  Everything is an 'experiment'.  Further down in the summary is a 'reference' list.

A "getting help" section makes sense to me. As it's currently implemented ("I need help") this article should probably replace "Getting Review" in the introduction since it handles all sorts of data related topics.

> 4.  Question:  'introduction.md'  The rest of the book is about being an Analyst.  Most of the experimenting stuff is about Needing To Make a Decision / being a PM.

This book is targeted at folks who are trying to make decisions with data. Articles targeted at non-analysts may belong somewhere else. It's still reasonable to keep these articles here while we find a new home, but these articles should probably be separated from the main document flow. 

> 5. Question:  I have a lot of 'wizards' / and 'case blocks'.  Mermaid?  Seem heavy and hard and weird. Examples:  "what do I need based on the evidence I want to collect / the decision I need to make".

"Choosing a Dataset" is a good example of guiding users through many options without graphics / in a narrative format.

**Comment by gregglind** [comment]
there are still loose ends here.

My brain is pretty fried trying to tie them up.  There is redundant text.

(I am also having trouble because the build time takes 60+ seconds, for every change, which makes it hard to check the book / croosslink).

**Comment by gregglind** [comment]
@harterrt

2nd pass.

1.  I feel like some of the content is redundant.
2.  Having the book take 60+ to build break live-reload
3.  The lunr indexer says it's now out of space.
4.  My brain is really fried on this.  Ideas and help welcome to rip out what can't ship.

File: experiments/deploying_code.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # How You Get Code To 100% Of Release
 
-_Warning: this is written by NOT a Release Manager, and has not (yet) been vetted.  Reflects state of the world on 2018/07/19._
+_Warning_: this is written by NOT a Release Manager, and has not (yet) been vetted.  Reflects state of the world on 2018/07/19.
```
Comment (biancadanforth) [inline experiments/deploying_code.md]: Feature roll-out seems outside of our team's wheelhouse? I argue this document is better to be written and maintained by Benson's team.

File: experiments/is_my_project_on_track.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+- are you hacking?
```
Comment (biancadanforth) [inline experiments/is_my_project_on_track.md]: The idea behind this document is valuable: Shield Studies can take on the order of months from start to finish. Here is a timeline breakdown and dependency graph of things you should be thinking about and when throughout the course of developing a Shield study.

File: experiments/need_shield_study.md [hunk 3]
```diff
@@ -1,6 +1,72 @@
-# Experimentation using Shield (Studies)
+# I Need a Shield Study
```
Comment (biancadanforth) [inline experiments/need_shield_study.md]: A lot of the contents of this document seem like they would be better placed in a higher level FAQ about experiments more broadly -- not about Shield studies.

File: experiments/need_shield_study.md [hunk 4]
```diff
@@ -50,12 +102,30 @@ I need evidence to know the next stage for my idea:
...
+
+Still here?
+
+That means, you have decided that you need to deploy
+ 
+- a realistic feature 
+- with complex interaction instrumentation
+- with complex ending criteria
+- to release users
+- using an addon.
```
Comment (biancadanforth) [inline experiments/need_shield_study.md]: Shield studies can deploy in pre-release channels, and while they usually are, they don't have to be add-on studies (ex: observational pref-flip studies).

File: experiments/shield_design_process.md [hunk 5]
```diff
@@ -4,39 +4,49 @@ The Mozilla Shield Program exists to help make decisions using Evidence.
 
 All engineering and process decisions in the Shield Program serve that goal.  
 
-Shield exists to:
+## Shield Decision Design Process
```
Comment (biancadanforth) [inline experiments/shield_design_process.md]: A walkthrough with a real life sample study would be very helpful here.

File: experiments/shield_study_engineering.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+
+## Topics of Shield Engineering
+
+
+### Do I need the Study Utils?
+
+[shield-studies-addon-utils](https://github.com/mozilla/shield-study-addon-utils)
+
+Study Utils makes these things easier.
```
Comment (biancadanforth) [inline experiments/shield_study_engineering.md]: Give some examples of mishaps when people have NOT used the utils. For example: There was a recent study that didn't [`setExperimentActive`](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/experiments.html?highlight=setexperimentactive#the-js-api), and so the analyst couldn't tell which users were actually enrolled in the experiment.

File: experiments/what_next.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+- Do you understand the decision you need to make?
+- Have you written a PHD?
+- Do you have a QA and Engineering acceptance plan?
+- Do you have 
+    - Firefox peers to review your code.  
+    - QA?  Can they run your QA plan?
+    - a PM?
+    - an analyst?  Have they approved your probes?
+    - a data reviewer?  Have they approved your probes?
+- Launch checklist:
```
Comment (biancadanforth) [inline experiments/what_next.md]: This is a great idea.

**Comment by harterrt** [comment]
Thanks Gregg. Comments inline:

>     I feel like some of the content is redundant.
>     My brain is really fried on this. Ideas and help welcome to rip out what can't ship

OK, I read through what you've written so far. There's a ton of great content here. Thank you for getting it down! It's going to take a while to complete all of the topics you've listed here, so I think you should start by committing **one** article that introduces shield. This will give us a foothold to build around. I've actually been amazed at how documentation starts to grow once you give it a nucleation point to start from.

Here's an outline of an example article. I offer this only because it seems like it would be helpful - not as a requirement. 

**Audience**: A new Firefox engineer going through onboarding or an existing engineer who knows nothing about how we experiment at Firefox. The reader probably has some reasonable conception about what an A/B experiment is.

**Style**: Prefer prose and a narrative. Checklists are good for monitoring ongoing work but the proposed document is meant to introduce a concept to a new engineer. Discard ideas that don't support this goal and audience (or save it somewhere for later).

**Suggested outline**:
* Intoduction
  * Goals for this document (After reading this document, you'll understand what the Shield program does and how it can help you make evidence driven decisions.)
  * TOC
* Shield
  * What is shield - Program for making evidence driven decisions
  * What is *not* shield - Deployment console
* Happy path for experimentation
  * Say Hi (Contact the shield team to introduce your project)
  * Start with a hypothesis (Why should you start with a hypothesis? What are the common failure cases?)
  * Define the users you want to target (How does this work, why do we do it)
  * Start with a survey 
  * Iterate over MVPs
* Getting help

---
>     Having the book take 60+ to build break live-reload

I'm not sure why this is taking so long on your machine. This branch builds in ~15s for me and a few other arbitrary users I asked. What does `gitbook --version` yield? 

>     The lunr indexer says it's now out of space.

I can't reproduce this either. Could there be additional files in your working directory confusing Gitbook? Do you have unpushed changes?

---

Thanks again for this work, Gregg! Let me know how I can help.

File: experiments/is_my_project_on_track.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        go do that.
+- Firefox Peer
+- PI-request
+    + requires a peer review
+    + qa plan
+
+
+Realistic date planning.
+
+- QA will discover problems.  They will be on inconvenient
```
Comment (pdehaan) [inline experiments/is_my_project_on_track.md]: Not sure if the "They" refers to "QA" or "problems" here, but either way it is valid and I love it.

File: experiments/shield_study_engineering.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+### General Shield Study Issues
+
+- Code requires Firefox Peer review.
+
+- beginings
+    + **First Install** is the first install different?
+        * introduction screen
+        * opt-in screen
+    + **Eligilibity criteria** for entering the study?
```
Comment (pdehaan) [inline experiments/shield_study_engineering.md]: typo: "Eligibility"

**Comment by gregglind** [comment]
@harterrt, I claim this is okay, other than spelling errors.

Thoughts welcome.

**Comment by gregglind** [comment]
@harterrt,

This has been sitting a while.

Should I just make a new PR?

There are still typos and words to add to the dictionary.  

The build cycle on my machine is making it really hard to debug and stamp out the last few errors.

Ideas:
- just move these docs over to Shield Stuides Addon Utils or Mana
- land a small page pointing to both here.  
- make a new PR?

I am wondering if maybe collaboration in markdown is just... hard.

**Comment by harterrt** [comment]
Hey @gregglind. Sorry for the slow reply, I was on PTO last week. 

> The build cycle on my machine is making it really hard to debug and stamp out the last few errors.

Can you try using mdbook (see #162)? That may fix your debug loop speed issues.

> Ideas:
> - just move these docs over to Shield Stuides Addon Utils or Mana
> - land a small page pointing to both here.  

This sounds reasonable to me. We can iterate on the docs here using the wiki page as a source.

I'm excited about this documentation! I've learned a lot about the shield program in reviewing. However, I do feel like the docs in their current form would still be confusing to someone who doesn't have much context on what shield is. I'll add some comments to the diff to clarify.

> I am wondering if maybe collaboration in markdown is just... hard.

It sounds like the problems you're having are related to Gitbook, not so much with markdown or Github. I'm sorry about this. It really sucks to have to use bad tools and it is directly against the goals for DTMO. I want to replace Gitbook with something faster, you can track progress in #162.

**Pull request closed without merging**